### PR TITLE
Add python-tqdm dependency for progress bars

### PR DESCRIPTION
**Pull request by cglwn**
See corresponding issue #15561 for motivation, use cases, and links to web listings for this package.

This is my first pull request into this repo. Please let me know if there's any changes or more information needed as I would like to get this in soon. :smile:

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
this looks good!

A few remarks:
- ros packages can be released only if its dependencies are available via the default package manager (e.g. apt for ubuntu and debian). As it stands, tqdm being available in apt only since yakkety, you will not be able to release a package depending on this in any of the existing ROS distributions (please see [REP3](http://www.ros.org/reps/rep-0003.html) for the list of targeted platform for each ROS diistribution)
  - Though you will be able to use this key to build from source

- We don't have any "alive" ROS Distribution targetting ubuntu versions older than Ubuntu Saucy (Indigo), if you are not planning on building from source on platform older than Saucy there is no need to add a rule for it

File: rosdep/python.yaml [hunk 1]
```diff
@@ -3281,6 +3281,52 @@ python-tornado-pip:
...
+    vivid:
+      pip:
+        packages: [tqdm]
+    wily:
+      pip:
+        packages: [tqdm]
+    xenial:
+      pip:
+        packages: [tqdm]
+    yakkity: [python-tqdm]
```
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: `yakkety`

**Review by Alessandro-Barbieri — CHANGES_REQUESTED** [review]

File: rosdep/python.yaml (hunk 1 above)
Comment (cglwn) [inline rosdep/python.yaml]: Fixed, thanks for catching :+1:!

**Comment by cglwn** [comment]
@mikaelarguedas I only added python-tqdm to make dependency management easier for source building and currently don't have any plans to release binaries in the near-term. The lack of ability to distribute it in binaries should be fine for my use case.

I've deleted the key-value pairs for the old Ubuntu versions in b92447c.

File: rosdep/python.yaml [hunk 2]
```diff
@@ -3281,6 +3281,31 @@ python-tornado-pip:
   ubuntu:
     pip:
       packages: [tornado]
+python-tqdm:
+  debian: [python-tqdm]
```
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: can you specify `stretch` and `buster`?

**Review by Alessandro-Barbieri — CHANGES_REQUESTED** [review]

File: rosdep/python.yaml (hunk 2 above)
Comment (cglwn) [inline rosdep/python.yaml]: :+1:

**Review by mikaelarguedas — APPROVED** [review]
lgtm, thanks @cglwn for the quick iteration

**Comment by cglwn** [comment]
Could this go in? I'm setting up CI for my project now which would be a lot simpler if this were in.

**Comment by mikaelarguedas** [comment]
Rosdep keys need 2 approvals to get merged, a member of the team will review and 'squash and merge' this PR soon

**Comment by cglwn** [comment]
Thanks for showing me the ropes and for the quick turnaround times @mikaelarguedas and @dirk-thomas!

**Pull request merged**